packages/datatrak-web/src/features/ProjectSelectForm.tsx

```diff
@@ -31,99 +31,100 @@ const ListWrapper = styled.div`
-  $hasAccess: boolean;
-}>`
-  color: ${({ theme, $hasAccess }) =>
-    $hasAccess ? theme.palette.text.primary : theme.palette.text.secondary};
-
-  .MuiSvgIcon-root {
-    color: ${({ theme }) => theme.palette.primary.main};
-  }
-`;
-
-const ListItem = ({ item, onSelect }) => {
-  const { name, selected, hasAccess } = item;
-  const onClick = () => {
-    onSelect(item);
-  };
-
-  const Item = (
-    <StyledListItem button onClick={onClick} selected={selected} $hasAccess={hasAccess}>
-      <IconWrapper>{!hasAccess && <Lock />}</IconWrapper>
-      {name}
-    </StyledListItem>
-  );
-
-  return hasAccess ? (
-    Item
-  ) : (
-    <Tooltip title="Request project access" arrow>
-      {Item}
-    </Tooltip>
-  );
-};
-
 interface ProjectSelectFormProps {
   projectId?: Project['id'];
   variant?: 'modal' | 'page';
   onClose: () => void;
+  onRequestAccess?: (projectCode: Project['code']) => void;
 }
 
 export const ProjectSelectForm = ({
   projectId,
   onClose,
   variant = 'page',
+  onRequestAccess,
 }: ProjectSelectFormProps) => {
   const navigate = useNavigate();
   const [selectedProjectId, setSelectedProjectId] = useState(projectId);
   const { data: projects, isLoading } = useProjects();
-  const { mutate, isLoading: isConfirming } = useEditUser(onClose);
+  const { mutate, isLoading: isConfirming, error } = useEditUser(onClose);
   const onConfirm = () => {
     mutate({ projectId: selectedProjectId! });
   };
 
-  const onRequestAccess = () => {
-    navigate(ROUTES.REQUEST_ACCESS);
-    if (variant === 'modal') {
-      onClose();
+  const handleRequestAccess = project => {
+    if (variant === 'modal' && onRequestAccess) {
+      onRequestAccess(project.code);
+    } else {
+      navigate({
+        pathname: ROUTES.REQUEST_ACCESS,
+        search: `?project=${project?.code}`,
+      });
     }
   };
 
   const onSelect = project => {
     if (project.hasAccess) {
       setSelectedProjectId(project.value);
     } else {
-      onRequestAccess();
+      handleRequestAccess(project);
     }
   };
 
-  const projectOptions = projects?.map(({ entityName, id }, index) => ({
-    name: entityName,
-    value: id,
-    selected: id === selectedProjectId,
-    // Todo: get hasAccess from api
-    hasAccess: index < 6,
-  }));
+  const getProjectIcon = (hasAccess: boolean, hasPendingAccess: boolean) => {
+    if (hasPendingAccess) return <WatchLater />;
+    if (!hasAccess) return <Lock />;
+    return null;
+  };
+
+  const getProjectTooltip = (hasAccess: boolean, hasPendingAccess: boolean) => {
+    if (!hasAccess && !hasPendingAccess) return 'Request project access';
+    return '';
+  };
+
+  const getProjectContent = (name: string, hasPendingAccess: boolean) => {
+    // customise the tooltip to appear over the project name if there is pending access, instead of the default of over the whole list item
+    if (hasPendingAccess)
+      return (
+        <Tooltip title="Approval in progress">
```

Comment:
Is it possible to consolidate the tooltips to either both use the ListItem tooltip prop or both be part of the content?

Reply:
The reason here is that the pending access state has the tooltip over just the text and the others have over the full width list item